pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -376,6 +383,26 @@ func (t *CycleNodeRequestTransitioner) transitionScalingUp() (reconcile.Result,
 		}
 	}
 
+	// Skip looping through nodes if no health checks need to be performed
+	if t.cycleNodeRequest.Spec.HealthChecks == nil {
```

Comment:
```suggestion
	if len(t.cycleNodeRequest.Spec.HealthChecks) == 0 {
```
I think is the more Go approach. Checks both nil and empty